security-privacy-questionnaire.md

```diff
@@ -0,0 +1,99 @@
+# Security and Privacy Questionnaire for the Push API
+
+This document answers the questions raised in the
+[Security and Privacy Questionnaire](https://www.w3.org/TR/security-privacy-questionnaire/)
+in context of the [Push API](https://w3c.github.io/push-api/).
+
+## 3.1 Does this specification deal with personally-identifiable information?
+
+The Push API issues a subscription that has an endpoint unique to the user. It
+persists until the subscription is cleared, either by the developer or by the
+user through Clear Site Data features provided by the user agent.
+
+[The specification states](https://w3c.github.io/push-api/#security-and-privacy-considerations)
+that endpoints must not be reused to avoid the creation of a persistent
+identifier.
+
+## 3.2 Does this specification deal with high-value data?
+
+The developer can specify an arbitrary payload that is to be included with the
+push message. Such a payload must be encrypted in accordance with
+[draft-ietf-webpush-encryption](https://tools.ietf.org/html/draft-ietf-webpush-encryption)
+to make sure that third parties, including the push service (a mandatory
+intermediary), cannot gain access to it.
+
+## 3.3 Does this specification introduce new state for an origin that persists across browsing sessions?
+
+Yes, a [push subscription](https://w3c.github.io/push-api/#push-subscription).
+
+## 3.4 Does this specification expose persistent, cross-origin state to the web?
+
+No.
+
+## 3.5 Does this specification expose any other data to an origin that it doesn’t currently have access to?
+
+No.
+
+## 3.6 Does this specification enable new script execution/loading mechanisms?
+
+Yes. The Push API enables developers to wake up their Service Worker remotely,
+without depending on the user to have an open window or tab to the origin, or,
+in some cases, without depending on the browser to be running at all.
+
+## 3.7 Does this specification allow an origin access to a user’s location?
+
+Potentially, through IP-to-location mechanisms when the Service Worker issues a
+fetch.
+
+## 3.8 Does this specification allow an origin access to sensors on a user’s device?
+
+No.
+
+## 3.9 Does this specification allow an origin access to aspects of a user’s local computing environment?
+
+Yes, the push service selected by the device. In some cases, most commonly on
+desktop platforms, a user agent includes a push service client separate from the
+push service made available by the operating system.
```

Comment:
I'm not sure what "access" means here.  The specification allows an origin to identify the push service.  I would say that the push service is selected by the user agent and that that might use the same push service as other applications.